compat/osdetect.h

```diff
@@ -66,7 +66,6 @@
 
 #elif defined(__MINGW32__) || defined(__MINGW__)
 #define _SQUID_MINGW_ 1
```

Comment:
Since we are changing the semantics of this macro, it may be a good time to also fix the undefined behavior its name triggers. In [C++](https://en.cppreference.com/w/c/language/identifier), "All identifiers that begin with an underscore followed by a capital letter" ... "are reserved and may not be declared in a program (doing so invokes undefined behavior)". Those UB identifiers include macros, and `_SQUID_MINGW_` is one of them.

Removing the leading underscore will introduce and inconsistency with other similar OS detection `_SQUID_` macros, so I do not insist on that removal, but we may have enough excuses to introduce that inconsistency in this case. Your call.

Reply:
This macro is not being touched by this PR, nor is its meaning changed in any way. This request is out of scope.